src/testRunner/unittests/helpers/tsserver.ts

```diff
@@ -355,31 +360,55 @@ export interface VerifyGetErrRequestBase {
     existingTimeouts?: boolean;
 }
 export interface VerifyGetErrRequest extends VerifyGetErrRequestBase {
-    files: readonly (string | File)[];
+    files: readonly (string | File | FileRangesRequestArgs)[];
     skip?: CheckAllErrors["skip"];
 }
 export function verifyGetErrRequest(request: VerifyGetErrRequest) {
     const { session, files } = request;
     session.executeCommandSeq<ts.server.protocol.GeterrRequest>({
         command: ts.server.protocol.CommandTypes.Geterr,
-        arguments: { delay: 0, files: files.map(filePath) },
+        arguments: {
+            delay: 0,
+            files: files.map(file => {
+                if (typeof file !== "string" && !(file as FileRangesRequestArgs).ranges) {
+                    return filePath(file as File);
+                }
+                return file as string | FileRangesRequestArgs;
+            }),
+        },
     });
     checkAllErrors(request);
 }
 
 interface SkipErrors {
     semantic?: true;
     suggestion?: true;
+    regionSemantic?: false;
 }
 export interface CheckAllErrors extends VerifyGetErrRequestBase {
     files: readonly any[];
     skip?: readonly (SkipErrors | undefined)[];
 }
 function checkAllErrors({ session, existingTimeouts, files, skip }: CheckAllErrors) {
-    for (let i = 0; i < files.length; i++) {
+    // Files with region will be checked first.
+    const fileSkips = files
+        .map((_file, i) => skip?.[i])
+        .sort((a, b) => {
+            if (a?.regionSemantic === false && b?.regionSemantic === undefined) return -1;
+            if (b?.regionSemantic === false && a?.regionSemantic === undefined) return 1;
+            return 0;
+        });
+    for (const fileSkip of fileSkips) {
+        // Run syntax check for next file
         session.host.runQueuedTimeoutCallbacks(existingTimeouts ? session.host.getNextTimeoutId() - 1 : undefined);
-        if (!skip?.[i]?.semantic) session.host.runQueuedImmediateCallbacks();
-        if (!skip?.[i]?.suggestion) session.host.runQueuedImmediateCallbacks();
+        if (fileSkip?.regionSemantic === false) {
+            // Run region check
+            session.host.runQueuedImmediateCallbacks();
+        }
+        // Run semantic check
+        if (!fileSkip?.semantic) session.host.runQueuedImmediateCallbacks();
+        // Run suggestion check
+        if (!fileSkip?.suggestion) session.host.runQueuedImmediateCallbacks();
     }

```

Comment:
Please also add the check after these to ensure that session contains keys only for open files in `semanticCheckPerformance`